deluge/core/alertmanager.py

```diff
@@ -41,13 +41,43 @@
 
 """
 
-from twisted.internet import reactor
+import threading
+
+from twisted.internet import reactor, threads
 
 import deluge.component as component
 from deluge._libtorrent import lt
 
 from deluge.log import LOG as log
 
+class AlertPoller(object):
+
+    def __init__(self, alert_manager):
+        self.alert_manager = alert_manager
+        self.running = False
+
+    def maybeUpdate(self):
+        if self.running:
+            self.alert_manager.update()
```

Comment:
Change this to call self.alert_manager.handle_alerts() directly.

Reply:
Done.